code/game/mecha/working/ripley.dm

```diff
@@ -54,22 +66,55 @@
 
 /obj/mecha/working/ripley/update_desc()
 	. = ..()
-	if(!hides) // Just in case if hides are somehow removed
+	if(!hides && !plates && !drake_hides) // Just in case if armor is removed
 		desc = initial(desc)
 		return
-	if(hides == 3)
+
+	// Goliath hides
+	if(hides == HIDES_COVERED_FULL)
 		desc = "Autonomous Power Loader Unit. It's wearing a fearsome carapace entirely composed of goliath hide plates - its pilot must be an experienced monster hunter."
 	else
 		desc = "Autonomous Power Loader Unit. Its armour is enhanced with some goliath hide plates."
 
+	// Metal plates
+	if(plates == PLATES_COVERED_FULL)
+		desc = "Autonomous Power Loader Unit. Its armor is completely lined with metal plating."
+	else
+		desc = "Autonomous Power Loader Unit. It's armor is reinforced with some metal plating."
+
+	// Drake hides
+	if(drake_hides == DRAKE_HIDES_COVERED_FULL)
+		desc = "Autonomous Power Loader Unit. Its every corner is covered in ancient hide, creating a powerful shield. The pilot of this exosuit must be prepared for battles on the level of legend."
+	if(drake_hides == DRAKE_HIDES_COVERED_MODERATE)
+		desc = "Autonomous Power Loader Unit. It`s armor is adorned with dragon hide plates, instilling fear in its enemies and guarding its pilot."
+	if(drake_hides == DRAKE_HIDES_COVERED_SLIGHT)
+		desc = "Autonomous Power Loader Unit. The armor of this exosuit only touches the mythical: a few plates of dragon hide adorn its plating like rare warrior trophies."
+
+
 /obj/mecha/working/ripley/update_overlays()
 	. = ..()
-	if(!hides)
-		return
-	if(hides == 3)
-		. += occupant ? "ripley-g-full" : "ripley-g-full-open"
-	else
-		. += occupant ? "ripley-g" : "ripley-g-open"
+// hides
+	if(hides)
+		if(hides == HIDES_COVERED_FULL)
+			. += occupant ? "ripley-g-full" : "ripley-g-full-open"
+		else
+			. += occupant ? "ripley-g" : "ripley-g-open"
+//plates
+	if(plates)
+		if(plates == PLATES_COVERED_FULL)
+			. += occupant ? "ripley-m-full" : "ripley-m-full-open"
+		else
+			. += occupant ? "ripley-m" : "ripley-m-open"
+//drake hides
+	if(drake_hides)
+		if(drake_hides == DRAKE_HIDES_COVERED_FULL)
+			. += occupant ? "ripley-d-full" : "ripley-d-full-open"
+		else if(drake_hides == DRAKE_HIDES_COVERED_MODERATE)
+			. += occupant ? "ripley-d-2" : "ripley-d-2-open"
```

Comment:
I can imagine "d" comes from Drake/Dragon, where does the "m" and "g" comes from?

Reply:
D - Dragon hide armor
M - Metal plate armor
G - Goliath hide armor